### PR TITLE
Prefer @wordpress/icons over dashicons in docs and tests

### PR DESCRIPTION
**Pull request by youknowriad**
Should be a harmless PR just updating icons used in docs and tests

### CODE REVIEW

File: packages/components/src/icon/README.md [hunk 1]
```diff
@@ -50,9 +58,7 @@ const MyIcon = () => (
 ```jsx
 import { Icon } from '@wordpress/components';
 
-const MyIcon = () => (
-	<Icon icon="screenoptions" className="example-class" />
-);
```
Comment (gziolo) [inline packages/components/src/icon/README.md]: I see still string here. Is it correct?

File: packages/edit-post/src/components/header/plugin-more-menu-item/index.js [hunk 2]
```diff
@@ -50,27 +51,28 @@ const PluginMoreMenuItem = ( { onClick = noop, ...props } ) => (
...
+ * import { more } from "@wordpress/icons";
  *
  * function onButtonClick() {
  * 	alert( 'Button clicked.' );
  * }
  *
  * const MyButtonMoreMenuItem = () => (
  * 	<PluginMoreMenuItem
- * 		icon="smiley"
+ * 		icon={more}
```
Comment (gziolo) [inline packages/edit-post/src/components/header/plugin-more-menu-item/index.js]: Missing spaces around the variable name.

File: packages/edit-post/src/components/header/plugin-more-menu-item/index.js [hunk 3]
```diff
@@ -50,27 +51,28 @@ const PluginMoreMenuItem = ( { onClick = noop, ...props } ) => (
...
+ * 	);
  * }
  * ```
  *
  * @example <caption>ESNext</caption>
  * ```jsx
  * // Using ESNext syntax
- * const { __ } = wp.i18n;
- * const { PluginMoreMenuItem } = wp.editPost;
+ * import { __ } from "@wordpress/i18n";
```
Comment (gziolo) [inline packages/edit-post/src/components/header/plugin-more-menu-item/index.js]: Here and in other places double quote is used. It should be single quote 😃

File: packages/edit-post/src/components/header/plugin-more-menu-item/test/__snapshots__/index.js.snap [hunk 4]
```diff
@@ -23,16 +23,16 @@ exports[`PluginMoreMenuItem renders menu item as button properly 1`] = `
     >
       <svg
         aria-hidden="true"
-        className="dashicon dashicons-smiley"
+        className="components-menu-items__item-icon"
         focusable="false"
         height={20}
         role="img"
-        viewBox="0 0 20 20"
```
Comment (gziolo) [inline packages/edit-post/src/components/header/plugin-more-menu-item/test/__snapshots__/index.js.snap]: Do we have changes that adjust CSS for a different size?

Ignore if yes. I’m on mobile and it’s hard to scroll 😅

**Review by gziolo — APPROVED** [review]

**Comment by gziolo** [comment]
Nice. What’s the plan for server- side registration of icons? It looks like we are deprecating strings as a way to list them.

File: packages/components/src/icon/README.md (hunk 1 above)
Comment (youknowriad) [reply packages/components/src/icon/README.md]: yes, this one will be kept for Backward compatibility (this component as well). I think we may remove the component from the handbook entirely at some point.

File: packages/edit-post/src/components/header/plugin-more-menu-item/test/__snapshots__/index.js.snap (hunk 4 above)
Comment (youknowriad) [reply packages/edit-post/src/components/header/plugin-more-menu-item/test/__snapshots__/index.js.snap]: I don't get the question :) I changed the icon here so it's a different viexbox

Comment (gziolo) [reply packages/edit-post/src/components/header/plugin-more-menu-item/test/__snapshots__/index.js.snap]: Lol, it's hard to explain. Dashicons have a different size than SVG icon but I think it's handled by the fact that you no longer use Dashicon here. Ignore me :)

Comment (youknowriad) [reply packages/edit-post/src/components/header/plugin-more-menu-item/test/__snapshots__/index.js.snap]: Yes, for dashicons that are still used we use a trick in viewport (-2) to center them properly even if they're smaller.

**Pull request merged**